statsd/statsd.go

```diff
@@ -68,14 +73,25 @@ var (
 	timingSuffix    = []byte("|ms")
 )
 
-// A Client is a handle for sending udp messages to dogstatsd.  It is safe to
+// A statsdWriter offers a standard interface regardless of the underlying
+// protocol. For now UDS and UPD writers are available.
+type statsdWriter interface {
+	Write(data []byte) error
```

Comment:
Hey @xvello and @truthbk 

Do you mind explaining why you chose a custom "Write" method and deviated from the std lib io.Writer interface?

I was fixing my [PR](https://github.com/DataDog/datadog-go/pull/31) and dismayed by this change.

If that was an oversight I can fix that in that PR as well.

Reply:
@cihangir we should make it implement `io.Writer` indeed, can you fix it in #31?